src/block/Block.php

```diff
@@ -427,14 +427,19 @@ public function getBreakInfo() : BlockBreakInfo{
 	 * @param Item[] &$returnedItems Items to be added to the target's inventory (or dropped, if full)
 	 */
 	public function onBreak(Item $item, ?Player $player = null, array &$returnedItems = []) : bool{
-		$world = $this->position->getWorld();
-		if(($t = $world->getTile($this->position)) !== null){
-			$t->onBlockDestroyed();
-		}
-		$world->setBlock($this->position, VanillaBlocks::AIR());
+		$this->position->getWorld()->setBlock($this->position, VanillaBlocks::AIR());
 		return true;
 	}
 
+	/**
+	 * Called when this block is destroyed either when a player breaks it or is hit by an explosion.
+	 */
+	public function onDestroy() : void{
```

Comment:
Like @dktapps I don't think that ``destroy`` terminology is appropriate in this case. Perhaps, as with the placement logic, an `onPostBreak` would be preferable? It would be less ambiguous and not so funny in the sense that the block would already have been destroyed.

Reply:
I'm not sure that's desirable. If the block was already destroyed, contents of stuff like tiles wouldn't be accessible.

Personally I would lean towards an `onExplode()` method or something similar. We already have the likes of `onIncinerate()` for fire. The problem is that this would require multiple handlers to be implemented for dealing with tile drops.